packages/adobe_analytics/ios/Classes/SwiftAdobeAnalyticsPlugin.swift

```diff
@@ -66,21 +71,27 @@ extension SwiftAdobeAnalyticsPlugin {
 //MARK: - AdobeAnalyticsProtocol conformance
 
 extension SwiftAdobeAnalyticsPlugin: AdobeAnalyticsProtocol {
-
+  
   public func trackAction(_ action: String, data: [String : String]?) {
     ACPCore.trackAction(action, data: data)
   }
-
+  
   public func trackState(_ state: String, data: [String : String]?) {
     ACPCore.trackState(state, data: data)
   }
-
+  
+  public func appendVisitorInfo(to url: URL, completion: @escaping (String) -> Void) {
+    ACPIdentity.append(to: url) { updatedURL in
+      completion(updatedURL!.absoluteString)
```

Comment:
Given Adobe's blackbox I wouldn't rely on `updatedURL` never being nil. (maybe if the SDK isn't properly initialised the returned URL's are nil, for example). So why not do `updatedURL?.absoluteString ?? url.absoluteString`?

Reply:
Good idea ; I'll update the PR 👍